### PR TITLE
rename to DucklingEntityExtractor

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- closes https://github.com/RasaHQ/rasa/issues/5661
- Rename `DucklingHTTPExtractor` to `DucklingEntityExtractor`

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by melindaloubser1** [comment]
Note: Rename in docs once new docs are merged

**Comment by melindaloubser1** [comment]
@erohmensing this can be reviewed / merged now that new structure is in and docs have been renamed too

File: rasa/nlu/constants.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 EXTRACTOR = "extractor"
 
-PRETRAINED_EXTRACTORS = {"DucklingHTTPExtractor", "SpacyEntityExtractor"}
+PRETRAINED_EXTRACTORS = {"DucklingEntityExtractor", "SpacyEntityExtractor"}
```
[the remark this answers is not in the document]
Comment (melindaloubser1) [reply rasa/nlu/constants.py:33]: should the extractor be changed to the new name first? or just add the HTTP one to the list of pretrained extractors in that case?

[the remark this answers is not in the document]
Comment (erohmensing) [reply rasa/nlu/constants.py:33]: Honestly i'm not sure what the best practice is here 🤔 not sure if it is better to do it like so:
```
PRETRAINED_EXTRACTORS = {
    "DucklingEntityExtractor", 
    "DucklingHTTPExtractor",  # for backwards compatibility when dumping Markdown
    "SpacyEntityExtractor", 
}
```
where its easier to notice and take out whenever we don't want to keep the compatibility, or whether we should
do 

```
extractors_to_ignore = PRETRAINED_EXTRACTORS + {"DucklingHTTPExtractor"})  # for backwards compatibility
if entity.get(EXTRACTOR) in extractors_to_ignore
```
in the method itself, where it's clearer why it's there. @m-vdb wdyt is the best practice here?

[the remark this answers is not in the document]
Comment (tmbo) [reply rasa/nlu/constants.py:33]: I think both ways are fine, for brevity I'd go with your first suggestion of just adding it and explaining the purpose of it being there. ideally also with a timeline on when it should be removed and what we need to change to make sure once we remove it things don't break (e.g. as you said, old predictions will still use that name and hence we need it to properly distinguish between entities we need to annotate and entities that are pretrained) - likely we won't ever remove it from that list

File: rasa/nlu/registry.py [hunk 2]
```diff
@@ -96,6 +96,15 @@
 def get_component_class(component_name: Text) -> Type["Component"]:
     """Resolve component name to a registered components class."""
 
+    if component_name == "DucklingHTTPExtractor":
+        raise_warning(
```
Comment (m-vdb) [inline rasa/nlu/registry.py:100]: should we use `raise_deprecation_warning()`?

**Review by erohmensing — APPROVED** [review]
I agree with maxime about using the new `raise_deprecation_warning`, otherwise looks good!

File: rasa/nlu/registry.py (hunk 2 above)
Comment (melindaloubser1) [reply rasa/nlu/registry.py:100]:
```suggestion
        raise_deprecation_warning(
```

**Pull request merged**